### PR TITLE
Add Pull Request Template

### PR DESCRIPTION
**Pull request by achiurizo**
Issue/Task Number: N/A

# Overview

This add a basic template for us to use for all our pull requests (from https://github.com/omisego/ewallet). We can continue iterating on this as needed.

# Changes

Adds a template file for us to use :)

# Implementation Details

N/A

# Usage

Using The PR flow in GitHub should bring up this template by default. Alternatively, if you use this via cli, this should also populate it from using `hub pull-request`

# Impact

N/A

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Issue/Task Number:
```
Comment (chrishunt) [inline .github/PULL_REQUEST_TEMPLATE.md]: Disappointed by the lack of 📎 or 📋 emoji

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# Overview
+
+Describe what your Pull Request is about in a few sentences. This is a global description that should tell people what your PR is all about. Don't go too much into details here.
```
Comment (chrishunt) [inline .github/PULL_REQUEST_TEMPLATE.md]: What do you think about chopping this down to just the first sentence? The rest of the description seems to be just rephrasing the same instructions using different words. 😃 

> Describe what your Pull Request is about in a few sentences

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+# Changes
+
+- Change 1
+- Change 2
+- ...
+
+# Implementation Details
```
Comment (chrishunt) [inline .github/PULL_REQUEST_TEMPLATE.md]: What do you think about merging the changes/implementation sections? They seem to be communicating the same pieces of information (how this thing was implemented, decisions made, trade-offs, etc..).

It looks like all of the recent PRs in the e-wallet repo are pretty much merging these sections as well.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+Describe your changes and implementation choices. More details make PRs easier to review.
+
+# Usage
```
Comment (chrishunt) [inline .github/PULL_REQUEST_TEMPLATE.md]: Should we rename this section to 'Testing'?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (achiurizo) [reply .github/PULL_REQUEST_TEMPLATE.md]: @chrishunt should have used the new 'suggest change' feature ;)

Comment (chrishunt) [reply .github/PULL_REQUEST_TEMPLATE.md]: Just saw that! I may try that next time 😄

**Pull request merged**